modules/internal/DefaultRectangular.chpl

```diff
@@ -1124,15 +1159,26 @@ module DefaultRectangular {
         if (rank == 1 && wantShiftedIndex) {
           return chpl__idxToInt(ind(1));
         } else {
-          var sum = 0:intIdxType;
+          //var sum = 0:intIdxType;
```

Comment:
Let's make this change controlled by a `config param usePollyArrayIndex = false`. That will leave it off by default (which is good for you, since then you don't have to ensure it is OK for benchmark performance when Polly is not used). We can turn it on for now with `chpl program.chpl -susePollyArrayIndex=true` and in the future when polly integration is enabled.